src/darwin/Framework/CHIP/ServerEndpoint/MTRServerAccessControl.h

```diff
@@ -0,0 +1,39 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRDefines.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * An access control implementation that looks at the MTRAccessGrants we have.
+ */
+@interface MTRServerAccessControl : NSObject
```

Comment:
Is this going to any instance methods in the future, or how is this used? Looks like this is just internal? I think simple free function would be sufficient here instead of a class.

Reply:
This is just internal, yes.  We might have instance methods at some point here, but for now I can do a free function, sure.

Reply:
Done.